# Data Cleaning

Research question: Is the proportion of current alcohol use different between male and female students?

Group variable: `WhatIsYourSex`  
Response variable: `CurrentAlcoholUse`

For `CurrentAlcoholUse`, code 1 means no current alcohol use, and codes 2–7 mean current alcohol use. The response variable is recoded as:

- 0 = did not currently use alcohol
- 1 = currently used alcohol

In [11]:
import pandas as pd
import numpy as np
from pathlib import Path

project_dir = Path("..")
raw_data_path = project_dir / "data" / "raw" / "YRBS_2007.csv"
processed_data_path = project_dir / "data" / "processed" / "cycle3_cleaned_alcohol_gender.csv"

df = pd.read_csv(raw_data_path)

df.shape

(14041, 103)

In [12]:
selected_vars = ["WhatIsYourSex", "CurrentAlcoholUse"]

df_selected = df[selected_vars].copy()

print("Sex variable:")
print(df_selected["WhatIsYourSex"].value_counts(dropna=False))

print("\nCurrent alcohol use variable:")
print(df_selected["CurrentAlcoholUse"].value_counts(dropna=False).sort_index())

Sex variable:
WhatIsYourSex
1.0    7036
2.0    6992
NaN      13
Name: count, dtype: int64

Current alcohol use variable:
CurrentAlcoholUse
1.0    6946
2.0    2735
3.0    1369
4.0     839
5.0     555
6.0     105
7.0     120
NaN    1372
Name: count, dtype: int64


In [13]:
df_clean = df_selected.dropna(subset=selected_vars).copy()

df_clean = df_clean[
    df_clean["WhatIsYourSex"].isin([1, 2]) &
    df_clean["CurrentAlcoholUse"].isin([1, 2, 3, 4, 5, 6, 7])
].copy()

df_clean["sex_group"] = df_clean["WhatIsYourSex"].map({
    1: "Female",
    2: "Male"
})

df_clean["current_alcohol_use"] = np.where(
    df_clean["CurrentAlcoholUse"] == 1,
    0,
    1
)

df_final = df_clean[["sex_group", "current_alcohol_use"]].copy()

df_final.head()

,sex_group,current_alcohol_use
3,Female,0
5,Female,0
6,Female,0
7,Female,0
8,Female,0


In [14]:
print("Sample size by group:")
print(df_final["sex_group"].value_counts())

print("\nAlcohol use by group:")
print(pd.crosstab(df_final["sex_group"], df_final["current_alcohol_use"]))

print("\nFinal shape:")
print(df_final.shape)

Sample size by group:
sex_group
Female    6425
Male      6234
Name: count, dtype: int64

Alcohol use by group:
current_alcohol_use     0     1
sex_group                      
Female               3561  2864
Male                 3381  2853

Final shape:
(12659, 2)


In [15]:
processed_data_path.parent.mkdir(parents=True, exist_ok=True)

df_final.to_csv(processed_data_path, index=False)

print("Cleaned data saved to:", processed_data_path)

Cleaned data saved to: ..\data\processed\cycle3_cleaned_alcohol_gender.csv


## Cleaning Summary

The original dataset had 14,041 rows. After removing missing values and keeping only valid codes for `WhatIsYourSex` and `CurrentAlcoholUse`, the final cleaned dataset had 12,659 rows.

The sex variable was recoded as:

- 1 = Female
- 2 = Male

The current alcohol use variable was recoded as:

- 0 = did not currently use alcohol
- 1 = currently used alcohol

The cleaned dataset was saved as `cycle3_cleaned_alcohol_gender.csv`.